core/src/network/handler/store.rs

```diff
@@ -39,6 +42,23 @@ pub fn handle_store_meta(dht_meta_data: DhtMetaData, context: Arc<Context>) {
                 }
             });
         }
+        "link_remove" => {
+            context.log("debug/net/handle: HandleStoreMeta: got LINK REMOVAL. processing...");
+            // TODO: do a loop on content once links properly implemented
+            assert_eq!(dht_meta_data.content_list.len(), 1);
+            let entry_with_header: EntryWithHeader = serde_json::from_str(
+                //should be careful doing slice access, it might panic
+                &serde_json::to_string(&dht_meta_data.content_list[0])
+                    .expect("dht_meta_data should be EntryWithHader"),
+            )
+            .expect("dht_meta_data should be EntryWithHader");
+            thread::spawn(move || {
+                match context.block_on(remove_link_workflow(&entry_with_header, &context.clone())) {
```

Comment:
You could make this much cleaner by using an `if let` expression

Reply:
Agreed! I think I found it structured like this and just added to it. In imo it would be a good candidate for a refactor since we are trying to get this story in for the closed alpha

Reply:
Then again it's not so hard to refactor a little as we go when we see easy improvements that would take almost no time...

Reply:
You are right, let me change that